modules/assim.sequential/R/Analysis_sda.R

```diff
@@ -151,128 +153,170 @@ GEF<-function(settings, Forecast, Observed, H, extraArg, nitr=50000, nburnin=100
+    for(i in 1:length(var.names)){
+      intervalX[which(startsWith(rownames(intervalX),
+                                 var.names[i])), ] <- matrix(c(as.numeric(settings$state.data.assimilation$state.variables[[i]]$min_value),
+                                                               as.numeric(settings$state.data.assimilation$state.variables[[i]]$max_value)),
+                                                             length(which(startsWith(rownames(intervalX),
+                                                                                     var.names[i]))),2,byrow = TRUE)
     }
-  }
-  
-  recompile <- extraArg$recompile
-  #browser()
-  ###TO DO: needs to recompile if there are new data sources
-  if(t == 1 | recompile){
-    #The purpose of this step is to impute data for mu.f 
-    #where there are zero values so that 
-    #mu.f is in 'tobit space' in the full model
-    constants.tobit2space <<- list(N = nrow(X),
-                                 J = length(mu.f))
-    
-    data.tobit2space <<- list(y.ind = x.ind,
-                            y.censored = x.censored,
-                            mu_0 = rep(0,length(mu.f)),
-                            lambda_0 = diag(length(mu.f),length(mu.f)+1),
-                            nu_0 = 3,
-                            wts = wts)#some measure of prior obs
-    
-    inits.tobit2space <<- list(pf = cov(X), muf = colMeans(X))
-    #set.seed(0)
-    #ptm <- proc.time()
-    
-    tobit2space_pred <<- nimbleModel(tobit2space.model, data = data.tobit2space,
-                                     constants = constants.tobit2space, inits = inits.tobit2space,
-                                     name = 'space')
-    ## Adding X.mod,q,r as data for building model.
-    conf_tobit2space <<- configureMCMC(tobit2space_pred, thin = 10, print=TRUE)
-    conf_tobit2space$addMonitors(c("pf", "muf","y.censored")) 
-    ## [1] conjugate_dmnorm_dmnorm sampler: X[1:5]
-    ## important!
-    ## this is needed for correct indexing later
-    samplerNumberOffset_tobit2space <<- length(conf_tobit2space$getSamplers())
     
+    #### These vectors are used to categorize data based on censoring from the interval matrix
+    x.ind <- x.censored <- matrix(NA, ncol=ncol(X), nrow=nrow(X))
     for(j in seq_along(mu.f)){
       for(n in seq_len(nrow(X))){
-        node <- paste0('y.censored[',n,',',j,']')
-        conf_tobit2space$addSampler(node, 'toggle', control=list(type='RW'))
-        ## could instead use slice samplers, or any combination thereof, e.g.:
-        ##conf$addSampler(node, 'toggle', control=list(type='slice'))
+        x.ind[n,j] <- as.numeric(X[n,j] > 0)
+        x.censored[n,j] <- as.numeric(ifelse(X[n,j] > intervalX[j,2], 0, X[n,j])) #
       }
     }
     
-    #conf_tobit2space$printSamplers()
-    
-    Rmcmc_tobit2space <<- buildMCMC(conf_tobit2space)
+    recompileTobit <- extraArg$recompileTobit
+    recompileGEF <- extraArg$recompileGEF
```

Comment:
`extraArg$recompileGEF` and ` extraArg$recompileTobit` are not set in the main `sda.enkf` function. This `extraArg$recompile` is set however. Either you can set those or replace em with the one that is set.